tic-tac-toe-app/Board.jsx

```diff
@@ -0,0 +1,87 @@
+// main board component
+function Board() {
+  // creating state for the board squares
+  const [boardSquares, setBoardSquares] = React.useState(
+    new Array(9).fill(null)
+  );
+
+  // state for the next turn
+  const [isXPlayer, setXPlayer] = React.useState(true);
+
+  // fucntion for when a sqaure is clicked
+  const whenClicked = index => {
+    // spread the items in boardSquare to new array to make a copy
+    const squares = [...boardSquares];
+    // check if square has a value
+    if (squares[index]) return;
+    // add next turn
+    squares[index] = isXPlayer ? 'X' : 'O';
+
+    // set the new state of the board squares
```

Comment:
The inline comments can help other developer better understand the intention of this code base. Groovy. 👍